app_dart/lib/src/service/luci_build_service.dart

```diff
@@ -0,0 +1,243 @@
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:cocoon_service/src/request_handling/exceptions.dart';
+import 'package:meta/meta.dart';
+
+import '../../cocoon_service.dart';
+import '../model/appengine/service_account_info.dart';
+import '../model/luci/buildbucket.dart';
+import 'buildbucket.dart';
+
+/// Class with utilities to interact with LUCI buildbucket to get, trigger
+/// and cancel builds for github repos. It uses [config.luciTryBuilders] to
+/// get the list of available builders.
+class LuciBuildService {
+  LuciBuildService(this.config, this.buildBucketClient, this.serviceAccount);
+
+  BuildBucketClient buildBucketClient;
+  Config config;
+  ServiceAccountInfo serviceAccount;
+  List<Status> failStatusList = <Status>[
+    Status.canceled,
+    Status.failure,
+    Status.infraFailure
+  ];
+
+  /// Returns a map of the BuildBucket builds for a given [repositoryName]
+  /// [prNumber] and [commitSha] using the [builderName] as key and [Build]
+  /// as value.
+  Future<Map<String, Build>> buildsForRepositoryAndPr(
+    String repositoryName,
+    int prNumber,
+    String commitSha,
+  ) async {
+    final BatchResponse batch =
+        await buildBucketClient.batch(BatchRequest(requests: <Request>[
+      Request(
+        searchBuilds: SearchBuildsRequest(
+          predicate: BuildPredicate(
+            builderId: const BuilderId(
+              project: 'flutter',
+              bucket: 'try',
+            ),
+            createdBy: serviceAccount.email,
+            tags: <String, List<String>>{
+              'buildset': <String>['pr/git/$prNumber'],
+              'github_link': <String>[
+                'https://github.com/flutter/$repositoryName/pull/$prNumber'
+              ],
+              'user_agent': const <String>['flutter-cocoon'],
+            },
+          ),
+        ),
+      ),
+      Request(
+        searchBuilds: SearchBuildsRequest(
+          predicate: BuildPredicate(
+            builderId: const BuilderId(
+              project: 'flutter',
+              bucket: 'try',
+            ),
+            tags: <String, List<String>>{
+              'buildset': <String>['pr/git/$prNumber'],
+              'user_agent': const <String>['recipe'],
+            },
+          ),
+        ),
+      ),
+    ]));
+    final Iterable<Build> builds = batch.responses
+        .map((Response response) => response.searchBuilds)
+        .expand(
+            (SearchBuildsResponse response) => response.builds ?? <Build>[]);
+    return Map<String, Build>.fromIterable(builds,
+        key: (dynamic b) => b.builderId.builder as String,
+        value: (dynamic b) => b as Build);
+  }
+
+  /// Schedules BuildBucket builds for a given [prNumber], [commitSha]
+  /// and repositoryName. It returns [true] if it was able to schedule
+  /// build or [false] otherwise.
+  Future<bool> scheduleBuilds({
```

Comment:
also commented here: https://github.com/flutter/cocoon/pull/760#issuecomment-636523657
What are the steps to test `write` api of LUCI?

Reply:
I'll comment in the PR but unit testing with mocked data is the best option for testing buildbucket related changes.